sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowTablesExec.scala

```diff
@@ -40,7 +40,7 @@ case class ShowTablesExec(
     val tables = catalog.listTables(namespace.toArray)
     tables.map { table =>
       if (pattern.map(StringUtils.filterPattern(Seq(table.name()), _).nonEmpty).getOrElse(true)) {
-        rows += toCatalystRow(table.namespace().quoted, table.name())
+        rows += toCatalystRow(table.namespace().quoted, table.name(), false)
```

Comment:
@cloud-fan It seems we should extend `catalog.listTables` so as output the `isTemporary`.

Reply:
This is v2 command, which can't use the built-in session catalog.

It does raise a question about whether SHOW TABLES should include temp views or not. We can investigate other databases. This can be done later.

Reply:
OK